packages/evolution-frontend/tests/ui-testing/SurveyObjectDetectors.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+// This file contains functions to detect survey objects from the survey update
+// data for playwright tests
+
+// Regexes for specific survey objects
+const uuidRegex = /[0-9a-f-]{36}/g
+const personObjectKeyRegex = /^responses\.household\.persons\.([0-9a-f-]{36})$/
+const activePersonKeyRegex = /^responses\._activePersonId$/
+const journeyObjectKeyRegex = /^responses\.household\.persons\.[0-9a-f-]{36}.journeys.[0-9a-f-]{36}$/
+const activeJourneyKeyRegex = /^responses\._activeJourneyId$/
+const visitedPlaceObjectKeyRegex = /^responses\.household\.persons\.[0-9a-f-]{36}.journeys.[0-9a-f-]{36}.visitedPlaces.[0-9a-f-]{36}$/
+const activeVisitedPlaceKeyRegex = /^responses\._activeVisitedPlaceId$/
+
+// Store the survey object IDs
+let personIds: string[] = [];
+let journeys: { [personId: string]: string[] } = {};
+let visitedPlaces: { [journeyId: string]: string[] } = {};
+// Store the active object IDs
+let activePersonId: string | undefined = undefined;
+let activeJourneyId: string | undefined = undefined;
+let activeVisitedPlaceId: string | undefined = undefined;
+
+// Replace ${personId[n]} with the actual personId
+const replaceWithPersonId = (str: string): string => {
+    let newString = str;
+    const personIdRegex = /\$\{personId\[(\d+)\]\}/g;
+    const matchGroups = str.match(personIdRegex);
+    if (matchGroups !== null) {
+        const personIndex = Number((matchGroups[0].match(/\d+/g)as any)[0]);
+        newString = str.replace(personIdRegex, personIds[personIndex]);
+    }
+    return newString;
+};
+
+const replaceWithActivePersonId = (str: string): string => {
+    const activePersonStr = '${activePersonId}';
+    return str.includes(activePersonStr) && typeof activePersonId === 'string' ? str.replace(activePersonStr, activePersonId) : str;
+}
+
+const replaceWithActiveJourneyId = (str: string): string => {
+    const activeJourneyStr = '${activeJourneyId}';
+    return str.includes(activeJourneyStr) && typeof activeJourneyId === 'string' ? str.replace(activeJourneyStr, activeJourneyId) : str;
+}
+
+const replaceWithActiveVisitedPlaceId = (str: string): string => {
+    const activeVisitedPlaceStr = '${activeVisitedPlaceId}';
+    return str.includes(activeVisitedPlaceStr) && typeof activeVisitedPlaceId === 'string' ? str.replace(activeVisitedPlaceStr, activeVisitedPlaceId) : str;
+}
```

Comment:
Any reason for 3 similar functions and not one? We can keep it like that, it's not a big deal. 

```
const replaceWithActivePersonId = (str: string): string => {
    const activePersonStr = '${activePersonId}';
    return str.includes(activePersonStr) && typeof activePersonId === 'string' ? str.replace(activePersonStr, activePersonId) : str;
}

const replaceWithActiveJourneyId = (str: string): string => {
    const activeJourneyStr = '${activeJourneyId}';
    return str.includes(activeJourneyStr) && typeof activeJourneyId === 'string' ? str.replace(activeJourneyStr, activeJourneyId) : str;
}

const replaceWithActiveVisitedPlaceId = (str: string): string => {
    const activeVisitedPlaceStr = '${activeVisitedPlaceId}';
    return str.includes(activeVisitedPlaceStr) && typeof activeVisitedPlaceId === 'string' ? str.replace(activeVisitedPlaceStr, activeVisitedPlaceId) : str;
}
```

Reply:
I'm currently adding the trips and segments. I think there are definite simplifications to do. That will be for the next next PR